src/Gui/ViewProviderPyImp.cpp

```diff
@@ -92,6 +92,39 @@ PyObject*  ViewProviderPy::addProperty(PyObject *args)
     return Py::new_reference_to(this);
 }
 
+PyObject*  ViewProviderPy::addLockedProperty(PyObject *args)
+{
+    char *sType,*sName=nullptr,*sGroup=nullptr,*sDoc=nullptr;
+    short attr=0;
+    std::string sDocStr;
+    PyObject *ro = Py_False, *hd = Py_False;
+    if (!PyArg_ParseTuple(args, "s|ssethO!O!", &sType,&sName,&sGroup,"utf-8",&sDoc,&attr,
+                          &PyBool_Type, &ro, &PyBool_Type, &hd))
+        return nullptr;
+
+    if (sDoc) {
+        sDocStr = sDoc;
+        PyMem_Free(sDoc);
+    }
+
+    App::Property* prop=nullptr;
+    try {
+        prop = getViewProviderPtr()->addDynamicProperty(sType,sName,sGroup,sDocStr.c_str(),attr,

```

Comment:
1) convert c_str to std::string
2) convert std::strong to c_str

Is there something I don't understand?

Reply:
I've copied the `addProperty` code above…

Reply:
@wwmayer do you know why you wrote this?